### PR TITLE
improve: use materialized view instead of base tables

### PR DESCRIPTION
**Pull request by amateima**
https://linear.app/uma/issue/ACX-109/get-all-referral-data-from-the-materialized-view

### CODE REVIEW

File: migrations/1658221793207-MaterializedView.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          d."depositTxHash",
+          d."sourceChainId",
+          d."destinationChainId",
+          d.amount,
+          t.symbol,
+          t.decimals,
+          d."depositorAddr",
+          d."referralAddress",
+          d."depositDate",
+          hmp.usd as "tokenUsdPrice",
```
Comment (amateima) [inline migrations/1658221793207-MaterializedView.ts:21]: add the token usd price as part of the materialized view

File: src/modules/referral/services/queries.ts [hunk 2]
```diff
@@ -47,49 +47,30 @@ export const getTotalReferralRewardsQuery = () => {
...
   return `select count(*) from (
     select distinct on (d."depositorAddr") d."depositorAddr"
-    from deposit d
-    where d."referralAddress" = $1 and
-          d."depositDate" is not null and
-          d."tokenId" is not null and
-          d."priceId" is not null and
-          d.status = 'filled'
+    from deposits_mv as d
+    where d."referralAddress" = $1
```
Comment (amateima) [inline src/modules/referral/services/queries.ts:51]: all the queries are basically the same, except that the target table is the `deposits_mv` materialized view

File: README.md [hunk 3]
```diff
@@ -18,3 +18,8 @@ docker-compose up
 ```shell script
 npm run lint
 ```
+
+## Deployment
+
+- `stage` branch is used to deploy it to the staging environment
```
Comment (nicholaspai) [inline README.md:24]: +1

File: migrations/1658221793207-MaterializedView.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          (d."realizedLpFeePctCapped" / power(10, 18)) * (d.amount / power(10, t.decimals)) * hmp.usd as "realizedLpFeeUsd",
```
Comment (nicholaspai) [inline migrations/1658221793207-MaterializedView.ts:22]: `realizedLpFeePctCapped` doesn't include `relayerFeePct` yet does it?

**Review by nicholaspai — APPROVED** [review]
Left a question

File: migrations/1658221793207-MaterializedView.ts (hunk 4 above)
Comment (amateima) [reply migrations/1658221793207-MaterializedView.ts:22]: @nicholaspai not yet. The bridge fee re-calculation is coming as a separate PR

**Pull request merged**